apps/server/src/modules/doc/manager.ts

```diff
@@ -357,4 +367,55 @@ export class DocManager implements OnModuleInit, OnModuleDestroy {
       },
     });
   }
+
+  /**
+   * deal with old records that has wrong guid format
+   * correct guid with `${non-wsId}:${variant}:${subId}` to `${variant}:${subId}`
+   *
+   * @TODO delete in next release
+   * @deprecated
+   */
+  protected async refreshDocGuid() {
+    let turn = 0;
+    let lastTurnCount = 100;
+    while (lastTurnCount === 100) {
+      const docs = await this.db.snapshot.findMany({
+        skip: turn * 100,
+        take: 100,
+        orderBy: {
+          createdAt: 'asc',
+        },
+      });
+
+      lastTurnCount = docs.length;
+      for (const doc of docs) {
+        // wrong guid
+        const parsed = DocID.parse(doc.id);
+        if (parsed && !parsed.isWorkspace) {
+          await this.db.snapshot.delete({
+            where: {
+              id_workspaceId: {
+                id: parsed.id,
+                workspaceId: doc.workspaceId,
+              },
+            },
+          });
+
+          await this.db.snapshot.update({
+            where: {
+              id_workspaceId: {
+                id: doc.id,
+                workspaceId: doc.workspaceId,
+              },
+            },
+            data: {
+              id: parsed.id,
+            },
+          });
+        }
```

Comment:
put in a transaction

Reply:
The delete statement is meaning to delete the unexpected data with wrong guid, like we fix `${ws}:space:${id}` to `space:${id}`, but the `space:${id}` should never exist before in the database, so in this way we could avoid primary key conflict.

Reply:
I checked my local database with some test workspace, somehow both `${ws}:space:${id}` and `space:${id}` existed, which is impossible in our codebase because guid itself is consistent once the doc is ever created.